### ISSUE TITLE
Some questions about supervision during training and validation

### ISSUE THREAD

**Issue by Cajeput** [nesl/radar-camera-fusion-depth#23]
Hello author, I have a few questions that I hope can be solved:

dgt mentioned in the paper refers to the lidar depth map provided by nuScenes, and dacc refers to the accumulated depth map. However, in the code, the lidar depth map provided by nuScenes is saved under `nuscenes_derived/lidar/`. However, `nuscenes_derived/ground_truth/` stores the accumulation depth map. 
Why is the ground truth in the paper and the ground truth in the code not the same thing? Why is the accumulation depth map not directly described as d_gt in the paper? Why is the lidar depth map provided by nuScenes described as d_lidar? There seems to be some contradiction between the description in the paper and the code.
Can it be understood that the ground truth in the code refers to the dacc in the paper, and the lidar in the code refers to the ground truth in the paper?

Looking forward to your reply. @thecyclone

**Comment by thecyclone** [comment]
Hi @Cajeput,

If you look at the [testing script](https://github.com/nesl/radar-camera-fusion-depth/blob/1e2cd8f4af5909bf2242acc6fb8caf031ccb7363/bash/run_fusionnet_nuscenes_test.sh#L14), you will notice that the ground truth being used is indeed the ground truth provided by nuScenes. All the test results are reported using this.

So yes, `dgt` in the paper refers to lidar in the code [For testing]

We chose to call the accumulated depth and interpolated depth as ground_truth since we were using them as the ground_truth during training.

**Comment by Cajeput** [comment]
Hello author, I have another question: In your paper, you mentioned the construction of binary classification tags for supervision through dacc; It is also mentioned that ygt is constructed by interpolating depth maps and is used for RadarNet's supervision. May I ask what are the differences and connections between these two labels? Why do you need binary classification tags built by dacc when you already have ygt monitoring?

**Comment by thecyclone** [comment]
Page 5 of the paper explains why we are doing this. Please refer to it.

**Comment by Cajeput** [comment]
Thank you. You mentioned that ygt is obtained by interpolating the depth map, but where RadarNet training parameters are set (Page 6:RadarNet (Stage-1)), you say that ygt is obtained by dacc, can you explain why?

**Comment by thecyclone** [comment]
My first answer in this thread is about Fusionnet supervision and not Radarnet supervision. I encourage you to read sections 4 and 5 of the paper to clear any confusion that you might have.

**Comment by Cajeput** [comment]
Hello, after carefully reading the code, I have found the following questions:
When training FusionNet, there are the following fragments in the `bash` script:
```
--train_ground_truth_path \
    training/nuscenes/nuscenes_train_ground_truth_interp.txt \
--train_lidar_map_path \
    training/nuscenes/nuscenes_train_ground_truth.txt \
```
The path of the interpolated depth map is saved in `nuscenes_train_ground_truth_interp.txt`, while the path of accumulating 161 frames of the accumulated depth map scanned by LiDAR is saved in `nuscenes_train_ground_truth. txt`. However, in train_fusionnet.py, these two command line parameters are assigned values to "`-- train_ground_truth-path`" and "`-- train_lidar_map_path`" respectively:
```
parser.add_argument('--train_ground_truth_path',
    type=str, required=True, help='Path to file that contains training paths for ground truth')
parser.add_argument('--train_lidar_map_path',
    type=str, required=True, help='Path to file that contains training paths for single lidar scan maps')
```
My question is: why is the accumulated depth map passed as a command line number to a single LiDAR scan` -- train_lidar_map_path`? (You mentioned in train_fusionnet.py that train_lidar_map_path is a single radar scan map)?

**Comment by thecyclone** [comment]
1. Please see the loss function of any network that you are concerned about in the paper -- Radarnet / Fusionnet
2. Look at the corresponding code -- Look at where we are calculating loss in the code
3. The corresponding values will help you understand what is what.

`ground_truth` and `ground_truth_interpolated` are basically two forms of dense supervisions available to us for training. The third type of supervision is called `lidar`. I encourage you to try training with a combination of any 2 of the 3. We chose the combination that delivered the best results for us empirically. 

Since it has been a while since we worked on this project, I am not certain what should be the correct combination in the bash file based on the paper but I think it will probably deliver the best results.
